pkg/auto/bms/logic.go

```diff
@@ -126,14 +159,14 @@ func analyseOccupancy(now time.Time, state *ReadState) (occupied, total, noRespo
 }
 
 // analyseTimeOfDay works out if we are currently within the deadband period, and when that might change.
-func analyseTimeOfDay(now time.Time, state *ReadState) (on bool, onStart, onEnd time.Time, changesIn time.Duration) {
+func analyseTimeOfDay(now time.Time, schedule []config.Range) (on bool, onStart, onEnd time.Time, changesIn time.Duration) {
```

Comment:
I got quite confused looking at this function's `onStart` return values, which returns when the currently active period will *next* start (e.g. tomorrow), instead of when the currently active period started in the past. i.e. `onStart` is after `onEnd`. This doesn't matter when the values are printed with time-of-day only (which they are). Still, this detail was not intuitive to me and I think it's worth a comment.